Mage.Server/src/main/java/mage/server/ChatSession.java

```diff
@@ -90,8 +90,8 @@ public void kill(UUID userId, DisconnectReason reason) {
                 String userName = clients.get(userId);
                 if (reason != DisconnectReason.LostConnection) { // for lost connection the user will be reconnected or session expire so no removeUserFromAllTablesAndChat of chat yet
                     final Lock w = lock.writeLock();
-                    w.lock();
                     try {
+                        w.lock();
                         clients.remove(userId);
                     } finally {
                         w.unlock();
```

Comment:
`w.lock();` must be outside from try-finally block... [look at discussion from previous pull request](https://github.com/magefree/mage/pull/4648/files#r176650560)